packages/shared/src/mock/utils.ts

```diff
@@ -31,22 +38,27 @@ export function assertMockAttribute<S extends Shape, T extends DType>(
   }
 }
 
-export function findMockEntity(path: string): Entity | undefined {
+export function findMockEntity(path: string): ProvidedEntity | undefined {
   assertAbsolutePath(path);
 
   if (path === '/') {
     return mockMetadata;
   }
 
-  const pathSegments = path.slice(1).split('/');
-  return pathSegments.reduce<Entity | undefined>(
-    (parentEntity, currSegment) => {
-      return parentEntity && isGroup(parentEntity) && hasChildren(parentEntity)
-        ? getChildEntity(parentEntity, currSegment)
-        : undefined;
-    },
-    mockMetadata
-  );
```

Comment:
I was struggling to fix the types in this `reduce`, so I refactored it with a recursion. The algorithm is similar to what is done in `HsdsApi.getEntity()`.